src/app/modules/shared/wallet_utils.nim

```diff
@@ -0,0 +1,116 @@
+import tables, sequtils, sugar
+
+import ../shared_models/[balance_item, currency_amount, token_item, token_model]
+
+import ../../../app_service/service/wallet_account/service as wallet_account_service
+import ../../../app_service/service/currency/dto as currency_dto
+
+import ../main/wallet_section/accounts/item as wallet_accounts_item
+import ../main/wallet_section/assets/item as wallet_assets_item
+import ../main/wallet_section/send/account_item as wallet_send_account_item
+import ../main/profile_section/wallet/accounts/item as wallet_settings_accounts_item
+import ../main/profile_section/wallet/accounts/[related_account_item, related_accounts_model]
```

Comment:
This seems a bit off to have dedicated module being dependent of shared and shared being dependent of dedicated module. Do you think we could avoid bi-directional dependency and have a factory which means we would have:

Factory depends of item
Item depends of shared?

Reply:
It's actually not a bidirectional dependencies in a logical way cause you're building an item based on dto, those actions are usually done in controllers/modules. It looks like bidirectional if you're looking at the file system like module requires something from shared location which requires something from module's location, but for example if we would move related items from modules' folders to `shared_models` location it would be a communication would be `module -> shared -> shared_models`.